apps/rush-lib/src/api/RepoStateFile.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
```

Comment:
This file feels like an internal implementation detail of Rush. If we're not going to expose it via the `rush-lib` API, maybe it should go under the "logic" folder instead of "api"